subprojects/platform-native/src/main/java/org/gradle/nativeplatform/CppSourceCompatibility.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.nativeplatform;
+
+import org.gradle.api.Incubating;
+
+/**
+ * Configures the C++ source compatibility that is to be used when compiling the C++ source files.
+ *
+ * @since 5.2
+ */
+@Incubating
+public enum CppSourceCompatibility {
+    /**
+     * Configures the source to be compatible with the 1998 C++ standard.
+     *
+     * <p>For example, this is equivalent to specifying the {@code -std=c++98} command-line flag for GCC.</p>
+     */
+    Cpp98,
+
+    /**
+     * Configures the source to be compatible with the compiler-specific extensions of the 1998 C++ standard.
+     *
+     * <p>For example, this is equivalent to specifying the {@code -std=gnu++98} command-line flag for GCC.</p>
+     */
+    Cpp98Extended,
```

Comment:
Good idea. Maybe it would be better to rename the `Extended` enumerations to `Gnu`? For example, either `Cpp98Gnu` or `GnuCpp98`. The latter seems to align better with how the CLI dialect option looks (e.g., `gnu++98`). WDYT?